libqtile/widget/mpris2widget.py

```diff
@@ -22,158 +22,152 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 
-from dbus_next.constants import MessageType  # type: ignore
+import asyncio
+from collections import defaultdict
+from string import Formatter
+
+import markupsafe
+from dbus_next import MessageType
+from dbus_next.aio import MessageBus
 
 from libqtile.log_utils import logger
 from libqtile.utils import add_signal_receiver
 from libqtile.widget import base
 
 
-class Mpris2(base._TextBox):
+class Mpris2(base._ScrollText):
     """An MPRIS 2 widget
 
     A widget which displays the current track/artist of your favorite MPRIS
     player. It should work with all MPRIS 2 compatible players which implement
-    a reasonably correct version of MPRIS, though I have only tested it with
-    audacious.  This widget scrolls the text if neccessary and information that
-    is displayed is configurable.
+    a reasonably correct version of MPRIS. This widget scrolls the text if
+    necessary and information that is displayed is configurable.
     """
-    orientations = base.ORIENTATION_HORIZONTAL
+
     defaults = [
         ('name', 'audacious', 'Name of the MPRIS widget.'),
-
         ('objname', 'org.mpris.MediaPlayer2.audacious',
             'DBUS MPRIS 2 compatible player identifier'
             '- Find it out with dbus-monitor - '
             'Also see: http://specifications.freedesktop.org/'
             'mpris-spec/latest/#Bus-Name-Policy'),
-
-        ('display_metadata', ['xesam:title', 'xesam:album', 'xesam:artist'],
-            'Which metadata identifiers to display. '
-            'See http://www.freedesktop.org/wiki/Specifications/mpris-spec/metadata/#index5h3 '
-            'for available values'),
-
-        ('scroll_chars', 30, 'How many chars at once to display.'),
-        ('scroll_interval', 0.5, 'Scroll delay interval.'),
-        ('scroll_wait_intervals', 8, 'Wait x scroll_interval before'
-            'scrolling/removing text'),
-
-        ('stop_pause_text', None, "Optional text to display when in the stopped/paused state"),
+        ('format', '{status_char}  <b>{title}</b> - {artist}',
+            'How to format the data.'
+            'Accepted keys are: "status_char" and Xesam properties.'
+            'https://www.freedesktop.org/wiki/Specifications/mpris-spec/metadata/'),
+        ('play_char', '', '"status_char" to display a song is playing.'),
+        ('pause_char', '', '"status_char" to display when a song is paused.'),
+        ('txt_inactive', '', 'Text to display when the player is inactive'),
+        ('scrolling_enabled', True, 'Wether to scroll the text.'),
```

Comment:
This looks like a bunch of breaking changes. Please either 1. implement a migration for these changes via qtile-migrate or 2. figure out how to make them non-breaking.

Reply:
I've adapted the signature so that it fits as much as possible the `Mpd2` widget syntax. I will take a look at `qtile migrate`, it seems awesome.

Reply:
The problem is that people who are existing users of the mpris widget will not necessarily be familiar with the mpd2 widget. It is best to make changes non-breaking if possible.

Reply:
Yes I get it. I will try to keep the old names when it can still make sense. However, what should I do to deprecate certain parameters smoothly ? Is there a guideline ?

Reply:
Yes, you'll need to implement a `qtile migrate` migration for anything you'd like to change.